src/main/scala/com/fulcrumgenomics/umi/DuplexConsensusCaller.scala

```diff
@@ -191,6 +223,19 @@ class DuplexConsensusCaller(override val readNamePrefix: String,
     }
   }
 
+  /** Split records into those that should make a single-end consensus read, first of pair consensus read,
+    * and second of pair consensus read, respectively.  For [[DuplexConsensusCaller]], no fragment reads should be
+    * given to this method, so none will be returned.
+    */
+  override protected def subGroupRecords(records: Seq[SamRecord]): (Seq[SamRecord], Seq[SamRecord], Seq[SamRecord]) = {
+    // NB: the input records should not have fragments
+    val (firstOfPair, secondOfPair) = records.partition { r =>
+      assert(r.paired, "Fragment reads should not be given to subGroupRecords in DuplexConsensusCaller.")
+      r.firstOfPair
+    }
+    (Seq.empty, firstOfPair, secondOfPair)
```

Comment:
If you really want every last iota of performance you might cache `val NoSamRecords = Seq.empty[SamRecord]` in the object and reference it here.  `Seq.empty` constructs a new one each time.